scalding-core/src/test/scala/com/twitter/scalding/mathematics/Matrix2Test.scala

```diff
@@ -221,6 +221,29 @@ class Matrix2Cosine(args: Args) extends Job(args) {
   cosine.write(TypedTsv[(Int, Int, Double)]("cosine"))
 }
 
+class Matrix2Normalize(args: Args) extends Job(args) {
+
+  import Matrix2._
+  import cascading.pipe.Pipe
+  import cascading.tuple.Fields
+  import com.twitter.scalding.TDsl._
+
+  val p1: Pipe = Tsv("mat1", ('x1, 'y1, 'v1)).read
+  val tp1 = p1.toTypedPipe[(Int, Int, Double)](('x1, 'y1, 'v1))
+  val mat1 = MatrixLiteral(tp1, NoClue)
+
+  // Now test for the case when value is Long type
+  val matL1Norm = mat1.rowL1Normalize
+  matL1Norm.write(TypedTsv[(Int, Int, Double)]("normalized"))
+
+  val p2: Pipe = Tsv("mat2", ('x2, 'y2, 'v2)).read // test Long type as value is OK
```

Comment:
let's use `TypedTsv` here and not use Pipe at all (just go right to `TypedPipe.from(TypedTsv[(Int, Int, Long)]("long_normalized")`)